scripts/versioning/check-version.ts

```diff
@@ -0,0 +1,54 @@
+// this script will be triggered from workflow actions when a new tag is push
+import {ethers} from 'hardhat'
+import {Contract} from 'ethers'
+import {ERC20Treasury} from '../../typechain'
+import {log} from '../../config/logging'
+import {deployContract} from '../common'
+
+const AddressZero = ethers.constants.AddressZero
+
+// for each deployable contract, check that the hardcoded version matches the tag version
+async function checkReleaseTag(contractName: string): Promise<boolean> {
+    const gitSourceTag = process.env.SOURCE_TAG
+    if (gitSourceTag) {
+        log.info(`git tag from workflow: ${gitSourceTag}`)
+    } else {
+        log.info(`git tag from workflow not defined!`)
+        process.exit(1)
+    }
+
+    log.info(
+        `Checking ${contractName} release against git release tag ${gitSourceTag}`
+    )
+
+    // how do we generically deploy contracts, with various constructor params
+    const contract: Contract = await deployContract<ERC20Treasury>(
+        contractName,
+        AddressZero,
+        AddressZero
+    )
+    log.info(`deployed contact at ${contract.address}`)
+
+    
+    const contractVersion: string =
```

Comment:
suggestion: use TypeChain to avoid the unsafe call.

TypeChain will generate a class for the `Version` abstract, there should be some way to have the contract retrieved as that type (it'll also get you familiar with TypeChain 😉 ).